validator/src/main/java/com/google/daq/mqtt/validator/validations/DiscoveryValidator.java

```diff
@@ -0,0 +1,161 @@
+package com.google.daq.mqtt.validator.validations;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.daq.mqtt.validator.CleanDateFormat;
+import com.google.daq.mqtt.validator.SequenceValidator;
+import com.google.daq.mqtt.validator.SkipTest;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import udmi.schema.DiscoveryConfig;
+import udmi.schema.DiscoveryEvent;
+import udmi.schema.FamilyDiscoveryConfig;
+import udmi.schema.FamilyDiscoveryState;
+
+/**
+ * Validation tests for discovery scan and enumeration capabilities.
+ */
+public class DiscoveryValidator extends SequenceValidator {
```

Comment:
May need to refactor these into smaller pieces in the future but ok to get started now